packages/core/src/common/resource.ts

```diff
@@ -224,29 +224,54 @@ export class DefaultResourceProvider {
 
 }
 
+export type ResourceInitializationOptions = Pick<Resource, 'autosaveable' | 'initiallyDirty' | 'readOnly'>
+    & { contents?: string | Promise<string>, onSave?: Resource['saveContents'] };
+
 export class MutableResource implements Resource {
-    protected contents: string = '';
+    protected contents: string | Promise<string>;
+
+    constructor(readonly uri: URI, protected options?: ResourceInitializationOptions) { }
```

Comment:
This makes all of the fields in this class effectively mutable.

Reply:
I consider that an advantage, given that it does have `Mutable` in its name.